File: jobs/2026-05-23__15-22-33/reminder_sleep_workflow_py__XvuoeB4/artifacts/user/myproject/activities.py
from temporalio import activity
import os

@activity.def
async def notify(message: str) -> str:
    log_file = "/workspace/reminder.log"
    # Ensure directory exists (though /workspace is requested to exist)
    os.makedirs(os.path.dirname(log_file), exist_ok=True)
    
    with open(log_file, "a") as f:
        f.write(f"Notified: {message}\n")
    
    return f"Notified: {message}"

File: jobs/2026-05-27__01-46-48/reminder_sleep_workflow_ts__hq4Kx2i/artifacts/user/myproject/src/activities.ts
import * as fs from 'fs';
import * as path from 'path';

const LOG_FILE = '/workspace/reminder.log';

/**
 * Notify activity: appends "Notified: <message>" to the log file and returns the string.
 */
export async function Notify(message: string): Promise<string> {
  const line = `Notified: ${message}`;

  // Ensure parent directory exists
  const dir = path.dirname(LOG_FILE);
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }

  fs.appendFileSync(LOG_FILE, line + '\n', 'utf8');

  return line;
}
